presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionWithIndexFunction.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.OperatorDependency;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import io.airlift.slice.Slice;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.common.function.OperatorType.EQUAL;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.util.Failures.internalError;
+
+@Description("Returns the position of the Nth occurrence of the given value in array (or 0 if not found)")
+@ScalarFunction("array_position")
+public class ArrayPositionWithIndexFunction
```

Comment:
Do we need to have 2 separate classes? Can we put all functions in one? Just curious.

Reply:
I tried this earlier. `FunctionsParserHelper.validateSignaturesCompatibility` mandates signatures of functions from one class to be the same. That's why I split it.

Reply:
Hmm, ok, try move the `@ScalarFunction` annotation to each implementation maybe. But that doesn't look ideal either. Don't have strong opinions here. Thanks!

Reply:
A better way would be to write a helper class and make both these classes call that one?

Reply:
@rongrong I tried moving `@ScalarFunction` but that unfortunately did not help. Therefore keeping two classes.
@kaikalur I am making only one of them as a source of truth and second just have one-liner function calls. This will hopefully keep it manageable.